File: stripe/src/test/java/com/stripe/android/model/ShippingMethodTest.java
package com.stripe.android.model;

import androidx.annotation.NonNull;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Test class for {@link ShippingMethod}
 */
public class ShippingMethodTest {
    private final static ShippingMethod SHIPPING_METHOD = createShippingMethod();

    @Test
    public void testEquals() {
        assertEquals(SHIPPING_METHOD, createShippingMethod());
    }

    @Test
    public void testHashcode() {
        assertEquals(SHIPPING_METHOD.hashCode(), createShippingMethod().hashCode());
    }

    @NonNull
    private static ShippingMethod createShippingMethod() {
        return new ShippingMethod("FedEx", "fedex", "Arrives tomorrow", 599, "USD");
    }
}

File: stripe/src/test/java/com/stripe/android/model/ShippingMethodTest.kt
package com.stripe.android.model

import com.stripe.android.utils.ParcelUtils
import kotlin.test.Test
import kotlin.test.assertEquals
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

/**
 * Test class for [ShippingMethod]
 */
@RunWith(RobolectricTestRunner::class)
class ShippingMethodTest {

    @Test
    fun testEquals() {
        assertEquals(SHIPPING_METHOD, createShippingMethod())
    }

    @Test
    fun testParcel() {
        assertEquals(SHIPPING_METHOD, ParcelUtils.create(SHIPPING_METHOD))
    }

    companion object {
        private val SHIPPING_METHOD = createShippingMethod()

        private fun createShippingMethod(): ShippingMethod {
            return ShippingMethod("FedEx", "fedex", 599, "USD", "Arrives tomorrow")
        }
    }
}
